sklearn/metrics/pairwise.py

```diff
@@ -738,9 +763,43 @@ def pairwise_distances_argmin(X, Y, *, axis=1, metric="euclidean", metric_kwargs
     if metric_kwargs is None:
         metric_kwargs = {}
 
-    return pairwise_distances_argmin_min(
-        X, Y, axis=axis, metric=metric, metric_kwargs=metric_kwargs
-    )[0]
+    X, Y = check_pairwise_arrays(X, Y)
+
+    if axis == 0:
+        X, Y = Y, X
+
+    if metric_kwargs is None:
+        metric_kwargs = {}
+
+    if PairwiseDistancesArgKmin.is_usable_for(X, Y, metric):
+        indices = PairwiseDistancesArgKmin.compute(
+            X=X,
+            Y=Y,
+            k=1,
+            metric=metric,
+            metric_kwargs=metric_kwargs,
+            strategy="auto",
+            return_distance=False,
+        )
+        indices = indices.flatten()
+    else:
+        # TODO: once PairwiseDistancesArgKmin supports sparse input matrices and 32 bit,
```

Comment:
Are there options to get 32bit to work besides `Tempita?

Reply:
Apart from duplication based on dtype in Cython, I hardly can see something else.

C++ template classes would come in handy here.

Reply:
I guess, we can start with Tempita first and see how this makes the code more complex.

What do the others think?

Reply:
If it works with fused types in Cython, that would be great. If not, C++ and Tempita a both fine for me.

Reply:
Fused types can't be used for attributes unfortunately (hence the need to use Tempita similarly to [what exists for `WeightVectors`](https://github.com/scikit-learn/scikit-learn/blob/0b570053ca2379b6786bd2c2a61b38ac4d347b7f/sklearn/utils/_weight_vector.pxd.tp#L26)).